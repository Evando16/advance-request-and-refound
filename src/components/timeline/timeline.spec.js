import React from 'react';
import { faUsers, faAsterisk, faConciergeBell } from '@fortawesome/free-solid-svg-icons';
import { act } from 'react-dom/test-utils';
import { render, unmountComponentAtNode } from 'react-dom';
import pretty from 'pretty';
import Timeline, { getCardIcon } from './timeline';

import requestTimelineData from './timeline-service';

const timelineDataMock = [
  {
    id: 545,
    cardDate: '01/04/2020',
    cardType: 'EXPENSE',
    expenseId: 545,
    invoiceDate: '01/04/2020',
    expenseTypeId: 5,
    expenseTypeCode: 'hotel-fee',
    expenseTypeIcon: 'concierge-bell',
    currencyId: 3,
    currencyCode: 'BRL',
    currencySymbol: 'R$',
    resourceUrl:
      'https://staging-backoffice-hotmart.s3.amazonaws.com/accountability/collaborator_6393359/accountability_742/expense_545_file_615_Captura%20de%20tela%20de%202019-11-24%2010-15-50.png',
    contentType: 'image/png',
    amountSpent: 11.11,
    amountTotal: 222.22,
    notes: 'sadasd',
    status: 'PENDING',
    updatedOn: 1585069977693,
    expenseEvaluation: null,
  },
];

jest.mock('./timeline-service');
let container = null;

beforeEach(() => {
  container = document.createElement('div');
  document.body.append(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('Unit tests', () => {
  it('should get icon to EVALUATION card', () => {
    expect(getCardIcon('EVALUATION')).toEqual(faUsers);
  });

  it('should get icon to ACCOUNTABILITY_SUBMITTED card', () => {
    expect(getCardIcon('ACCOUNTABILITY_SUBMITTED')).toEqual(faAsterisk);
  });

  it('should get icon to ACCOUNTABILITY_CREATED card', () => {
    expect(getCardIcon('ACCOUNTABILITY_CREATED')).toEqual(faAsterisk);
  });

  it('should get icon to EXPENSE card', () => {
    expect(getCardIcon('EXPENSE')).toEqual(faConciergeBell);
  });

  it('should get null icon to ivalid card type', () => {
    expect(getCardIcon('invalid')).toEqual(null);
  });

  it('should render no result while requesting timeline data', async () => {
    requestTimelineData.mockImplementation(() => Promise.resolve([]));

    await act(async () => {
      render(<Timeline />, container);
    });

    expect(container.textContent).toEqual('No timeline content =(');
  });

  it('should render timeline', async () => {
    requestTimelineData.mockImplementation(() => Promise.resolve(timelineDataMock));

    await act(async () => {
      render(<Timeline />, container);
    });

    expect(pretty(container.innerHTML)).toMatchInlineSnapshot(`
      "<div class=\\"MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded\\" style=\\"margin-top: 24px; padding: 16px;\\">
        <div><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"concierge-bell\\" class=\\"svg-inline--fa fa-concierge-bell fa-w-16 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\">
            <path fill=\\"currentColor\\" d=\\"M288 130.54V112h16c8.84 0 16-7.16 16-16V80c0-8.84-7.16-16-16-16h-96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h16v18.54C115.49 146.11 32 239.18 32 352h448c0-112.82-83.49-205.89-192-221.46zM496 384H16c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h480c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16z\\"></path>
          </svg><span>01/04/2020</span></div>
        <div><span>TIPO</span><span>Aprovação da Solicitação Henrique Elias</span></div>
        <div><span>VALOR</span><span>11.11</span><span>222.22</span></div>
        <div><span>STATUS</span><span>PENDING</span><span>sadasd</span></div>
        <div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"receipt\\" class=\\"svg-inline--fa fa-receipt fa-w-12 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 384 512\\"><path fill=\\"currentColor\\" d=\\"M358.4 3.2L320 48 265.6 3.2a15.9 15.9 0 0 0-19.2 0L192 48 137.6 3.2a15.9 15.9 0 0 0-19.2 0L64 48 25.6 3.2C15-4.7 0 2.8 0 16v480c0 13.2 15 20.7 25.6 12.8L64 464l54.4 44.8a15.9 15.9 0 0 0 19.2 0L192 464l54.4 44.8a15.9 15.9 0 0 0 19.2 0L320 464l38.4 44.8c10.5 7.9 25.6.4 25.6-12.8V16c0-13.2-15-20.7-25.6-12.8zM320 360c0 4.4-3.6 8-8 8H72c-4.4 0-8-3.6-8-8v-16c0-4.4 3.6-8 8-8h240c4.4 0 8 3.6 8 8v16zm0-96c0 4.4-3.6 8-8 8H72c-4.4 0-8-3.6-8-8v-16c0-4.4 3.6-8 8-8h240c4.4 0 8 3.6 8 8v16zm0-96c0 4.4-3.6 8-8 8H72c-4.4 0-8-3.6-8-8v-16c0-4.4 3.6-8 8-8h240c4.4 0 8 3.6 8 8v16z\\"></path></svg>Ver nota fiscal</span><span class=\\"MuiTouchRipple-root\\"></span></button></div>
      </div>"
    `);
  });
});
