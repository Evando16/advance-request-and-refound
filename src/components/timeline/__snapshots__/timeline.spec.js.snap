// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineComponent Snapshot tests should render timeline with all cards type 1`] = `
"<div data-testid=\\"solicitation__timeline\\" class=\\"TimelineContainier-bvCAbJ bTSqTF\\">
  <div class=\\"TimelineSubContainier-kqcMpN hzuChq\\">
    <div width=\\"12%\\" class=\\"TimelineQuadrant-fwbGLn TimelineIdentificationQuadrant-bswpNF hBbppX lbXdqa\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"concierge-bell\\" class=\\"svg-inline--fa fa-concierge-bell fa-w-16 TimelineIcon-cVZDCs eAuiWh\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\" backgroundcolor=\\"#e7f2fd\\" color=\\"#0f68bd\\">
        <path fill=\\"currentColor\\" d=\\"M288 130.54V112h16c8.84 0 16-7.16 16-16V80c0-8.84-7.16-16-16-16h-96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h16v18.54C115.49 146.11 32 239.18 32 352h448c0-112.82-83.49-205.89-192-221.46zM496 384H16c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h480c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16z\\"></path>
      </svg><span class=\\"TimelineSubDescription-UGctF iVIATh\\">04/01/2020</span></div>
    <div class=\\"TimelineQuadrant-fwbGLn jcrlXN\\"><span class=\\"TimelineLabel-bjBMFt hmpenS\\">ACTION</span><span class=\\"TimelineDesription-AVioK gyAJrn\\">Hotel</span></div>
    <div width=\\"20%\\" class=\\"TimelineQuadrant-fwbGLn gjeHFt\\"><span class=\\"TimelineLabel-bjBMFt hmpenS\\">VALUE</span><span class=\\"TimelineDesription-AVioK gyAJrn\\"><span class=\\"TimelineCurrencyLabel-gXIWDu cZUiCN\\"></span><span>11,11</span></span><span class=\\"TimelineSubDescription-UGctF iVIATh\\"><span class=\\"TimelineCurrencyLabel-gXIWDu cZUiCN\\">Receipt total value:</span><span class=\\"TimelineCurrencyLabel-gXIWDu cZUiCN\\"></span><span>222,22</span></span></div>
  </div>
  <div class=\\"TimelineSubContainier-kqcMpN kFKWsF\\">
    <div width=\\"15%\\" class=\\"TimelineQuadrant-fwbGLn FbnPL\\"><span class=\\"TimelineLabel-bjBMFt hmpenS\\">STATUS</span><span color=\\"#053d4e\\" class=\\"TimelineStatus-eaSVrj cGOjDd\\">Pending</span><span class=\\"TimelineSubDescription-UGctF iVIATh\\">sadasd</span></div>
    <div class=\\"TimelineQuadrantReceipt-bEVjwf iEsLpf\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"receipt\\" class=\\"svg-inline--fa fa-receipt fa-w-12 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 384 512\\">
        <path fill=\\"currentColor\\" d=\\"M358.4 3.2L320 48 265.6 3.2a15.9 15.9 0 0 0-19.2 0L192 48 137.6 3.2a15.9 15.9 0 0 0-19.2 0L64 48 25.6 3.2C15-4.7 0 2.8 0 16v480c0 13.2 15 20.7 25.6 12.8L64 464l54.4 44.8a15.9 15.9 0 0 0 19.2 0L192 464l54.4 44.8a15.9 15.9 0 0 0 19.2 0L320 464l38.4 44.8c10.5 7.9 25.6.4 25.6-12.8V16c0-13.2-15-20.7-25.6-12.8zM320 360c0 4.4-3.6 8-8 8H72c-4.4 0-8-3.6-8-8v-16c0-4.4 3.6-8 8-8h240c4.4 0 8 3.6 8 8v16zm0-96c0 4.4-3.6 8-8 8H72c-4.4 0-8-3.6-8-8v-16c0-4.4 3.6-8 8-8h240c4.4 0 8 3.6 8 8v16zm0-96c0 4.4-3.6 8-8 8H72c-4.4 0-8-3.6-8-8v-16c0-4.4 3.6-8 8-8h240c4.4 0 8 3.6 8 8v16z\\"></path>
      </svg><a href=\\"https://staging-backoffice-hotmart.s3.amazonaws.com/accountability/collaborator_6393359/accountability_742/expense_545_file_615_Captura%20de%20tela%20de%202019-11-24%2010-15-50.png\\" class=\\"TimelineReceiptLink-kgwDmf eWyroT\\">Receipt</a></div>
  </div>
</div>
<div data-testid=\\"solicitation__timeline\\" class=\\"TimelineContainier-bvCAbJ bTSqTF\\">
  <div class=\\"TimelineSubContainier-kqcMpN hzuChq\\">
    <div width=\\"12%\\" class=\\"TimelineQuadrant-fwbGLn TimelineIdentificationQuadrant-bswpNF hBbppX lbXdqa\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"utensils\\" class=\\"svg-inline--fa fa-utensils fa-w-13 TimelineIcon-cVZDCs eAuiWh\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 416 512\\" backgroundcolor=\\"#e7f2fd\\" color=\\"#0f68bd\\">
        <path fill=\\"currentColor\\" d=\\"M207.9 15.2c.8 4.7 16.1 94.5 16.1 128.8 0 52.3-27.8 89.6-68.9 104.6L168 486.7c.7 13.7-10.2 25.3-24 25.3H80c-13.7 0-24.7-11.5-24-25.3l12.9-238.1C27.7 233.6 0 196.2 0 144 0 109.6 15.3 19.9 16.1 15.2 19.3-5.1 61.4-5.4 64 16.3v141.2c1.3 3.4 15.1 3.2 16 0 1.4-25.3 7.9-139.2 8-141.8 3.3-20.8 44.7-20.8 47.9 0 .2 2.7 6.6 116.5 8 141.8.9 3.2 14.8 3.4 16 0V16.3c2.6-21.6 44.8-21.4 48-1.1zm119.2 285.7l-15 185.1c-1.2 14 9.9 26 23.9 26h56c13.3 0 24-10.7 24-24V24c0-13.2-10.7-24-24-24-82.5 0-221.4 178.5-64.9 300.9z\\"></path>
      </svg><span class=\\"TimelineSubDescription-UGctF iVIATh\\">03/25/2020</span></div>
    <div class=\\"TimelineQuadrant-fwbGLn jcrlXN\\"><span class=\\"TimelineLabel-bjBMFt hmpenS\\">ACTION</span><span class=\\"TimelineDesription-AVioK gyAJrn\\">Food</span></div>
    <div width=\\"20%\\" class=\\"TimelineQuadrant-fwbGLn gjeHFt\\"><span class=\\"TimelineLabel-bjBMFt hmpenS\\">VALUE</span><span class=\\"TimelineDesription-AVioK gyAJrn\\"><span class=\\"TimelineCurrencyLabel-gXIWDu cZUiCN\\"></span><span>1,11</span></span><span class=\\"TimelineSubDescription-UGctF iVIATh\\"><span class=\\"TimelineCurrencyLabel-gXIWDu cZUiCN\\">Receipt total value:</span><span class=\\"TimelineCurrencyLabel-gXIWDu cZUiCN\\"></span><span>2,22</span></span></div>
  </div>
  <div class=\\"TimelineSubContainier-kqcMpN kFKWsF\\">
    <div width=\\"15%\\" class=\\"TimelineQuadrant-fwbGLn FbnPL\\"><span class=\\"TimelineLabel-bjBMFt hmpenS\\">STATUS</span><span color=\\"#053d4e\\" class=\\"TimelineStatus-eaSVrj cGOjDd\\">Pending</span><span class=\\"TimelineSubDescription-UGctF iVIATh\\">dsfsdsffds</span></div>
    <div class=\\"TimelineQuadrantReceipt-bEVjwf iEsLpf\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"receipt\\" class=\\"svg-inline--fa fa-receipt fa-w-12 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 384 512\\">
        <path fill=\\"currentColor\\" d=\\"M358.4 3.2L320 48 265.6 3.2a15.9 15.9 0 0 0-19.2 0L192 48 137.6 3.2a15.9 15.9 0 0 0-19.2 0L64 48 25.6 3.2C15-4.7 0 2.8 0 16v480c0 13.2 15 20.7 25.6 12.8L64 464l54.4 44.8a15.9 15.9 0 0 0 19.2 0L192 464l54.4 44.8a15.9 15.9 0 0 0 19.2 0L320 464l38.4 44.8c10.5 7.9 25.6.4 25.6-12.8V16c0-13.2-15-20.7-25.6-12.8zM320 360c0 4.4-3.6 8-8 8H72c-4.4 0-8-3.6-8-8v-16c0-4.4 3.6-8 8-8h240c4.4 0 8 3.6 8 8v16zm0-96c0 4.4-3.6 8-8 8H72c-4.4 0-8-3.6-8-8v-16c0-4.4 3.6-8 8-8h240c4.4 0 8 3.6 8 8v16zm0-96c0 4.4-3.6 8-8 8H72c-4.4 0-8-3.6-8-8v-16c0-4.4 3.6-8 8-8h240c4.4 0 8 3.6 8 8v16z\\"></path>
      </svg><a href=\\"https://staging-backoffice-hotmart.s3.amazonaws.com/accountability/collaborator_6393359/accountability_742/expense_544_file_614_Captura%20de%20tela%20de%202019-11-21%2009-36-41.png\\" class=\\"TimelineReceiptLink-kgwDmf eWyroT\\">Receipt</a></div>
  </div>
</div>
<div data-testid=\\"solicitation__timeline\\" class=\\"TimelineContainier-bvCAbJ bTSqTF\\">
  <div class=\\"TimelineSubContainier-kqcMpN hzuChq\\">
    <div width=\\"12%\\" class=\\"TimelineQuadrant-fwbGLn TimelineIdentificationQuadrant-bswpNF hBbppX lbXdqa\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"asterisk\\" class=\\"svg-inline--fa fa-asterisk fa-w-16 TimelineIcon-cVZDCs ijgjbN\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\" backgroundcolor=\\"#ecf8f8\\" color=\\"#03a046\\">
        <path fill=\\"currentColor\\" d=\\"M478.21 334.093L336 256l142.21-78.093c11.795-6.477 15.961-21.384 9.232-33.037l-19.48-33.741c-6.728-11.653-21.72-15.499-33.227-8.523L296 186.718l3.475-162.204C299.763 11.061 288.937 0 275.48 0h-38.96c-13.456 0-24.283 11.061-23.994 24.514L216 186.718 77.265 102.607c-11.506-6.976-26.499-3.13-33.227 8.523l-19.48 33.741c-6.728 11.653-2.562 26.56 9.233 33.037L176 256 33.79 334.093c-11.795 6.477-15.961 21.384-9.232 33.037l19.48 33.741c6.728 11.653 21.721 15.499 33.227 8.523L216 325.282l-3.475 162.204C212.237 500.939 223.064 512 236.52 512h38.961c13.456 0 24.283-11.061 23.995-24.514L296 325.282l138.735 84.111c11.506 6.976 26.499 3.13 33.227-8.523l19.48-33.741c6.728-11.653 2.563-26.559-9.232-33.036z\\"></path>
      </svg><span class=\\"TimelineSubDescription-UGctF iVIATh\\">03/24/2020</span></div>
    <div class=\\"TimelineQuadrant-fwbGLn jcrlXN\\"><span class=\\"TimelineLabel-bjBMFt hmpenS\\">ACTION</span><span class=\\"TimelineDesription-AVioK gyAJrn\\">Solicitation submitted by Henrique Elias</span></div>
  </div>
  <div class=\\"TimelineSubContainier-kqcMpN kFKWsF\\"></div>
</div>
<div data-testid=\\"solicitation__timeline\\" class=\\"TimelineContainier-bvCAbJ bTSqTF\\">
  <div class=\\"TimelineSubContainier-kqcMpN hzuChq\\">
    <div width=\\"12%\\" class=\\"TimelineQuadrant-fwbGLn TimelineIdentificationQuadrant-bswpNF hBbppX lbXdqa\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"users\\" class=\\"svg-inline--fa fa-users fa-w-20 TimelineIcon-cVZDCs ebPbWO\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 640 512\\" backgroundcolor=\\"#f0f3f7\\" color=\\"#5f6772\\">
        <path fill=\\"currentColor\\" d=\\"M96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm448 0c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm32 32h-64c-17.6 0-33.5 7.1-45.1 18.6 40.3 22.1 68.9 62 75.1 109.4h66c17.7 0 32-14.3 32-32v-32c0-35.3-28.7-64-64-64zm-256 0c61.9 0 112-50.1 112-112S381.9 32 320 32 208 82.1 208 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zm-223.7-13.4C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z\\"></path>
      </svg><span class=\\"TimelineSubDescription-UGctF iVIATh\\">03/24/2020</span></div>
    <div class=\\"TimelineQuadrant-fwbGLn jcrlXN\\"><span class=\\"TimelineLabel-bjBMFt hmpenS\\">ACTION</span><span class=\\"TimelineDesription-AVioK gyAJrn\\">Solicitation pendding with Henrique Elias</span></div>
  </div>
  <div class=\\"TimelineSubContainier-kqcMpN kFKWsF\\">
    <div width=\\"15%\\" class=\\"TimelineQuadrant-fwbGLn FbnPL\\"><span class=\\"TimelineLabel-bjBMFt hmpenS\\">STATUS</span><span color=\\"#053d4e\\" class=\\"TimelineStatus-eaSVrj cGOjDd\\">Pending</span><span class=\\"TimelineSubDescription-UGctF iVIATh\\"></span></div>
  </div>
</div>
<div data-testid=\\"solicitation__timeline\\" class=\\"TimelineContainier-bvCAbJ bTSqTF\\">
  <div class=\\"TimelineSubContainier-kqcMpN hzuChq\\">
    <div width=\\"12%\\" class=\\"TimelineQuadrant-fwbGLn TimelineIdentificationQuadrant-bswpNF hBbppX lbXdqa\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"users\\" class=\\"svg-inline--fa fa-users fa-w-20 TimelineIcon-cVZDCs ebPbWO\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 640 512\\" backgroundcolor=\\"#f0f3f7\\" color=\\"#5f6772\\">
        <path fill=\\"currentColor\\" d=\\"M96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm448 0c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm32 32h-64c-17.6 0-33.5 7.1-45.1 18.6 40.3 22.1 68.9 62 75.1 109.4h66c17.7 0 32-14.3 32-32v-32c0-35.3-28.7-64-64-64zm-256 0c61.9 0 112-50.1 112-112S381.9 32 320 32 208 82.1 208 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zm-223.7-13.4C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z\\"></path>
      </svg><span class=\\"TimelineSubDescription-UGctF iVIATh\\">03/24/2020</span></div>
    <div class=\\"TimelineQuadrant-fwbGLn jcrlXN\\"><span class=\\"TimelineLabel-bjBMFt hmpenS\\">ACTION</span><span class=\\"TimelineDesription-AVioK gyAJrn\\">Solicitation cancelled by Henrique Elias</span></div>
  </div>
  <div class=\\"TimelineSubContainier-kqcMpN kFKWsF\\">
    <div width=\\"15%\\" class=\\"TimelineQuadrant-fwbGLn FbnPL\\"><span class=\\"TimelineLabel-bjBMFt hmpenS\\">STATUS</span><span color=\\"#f44336\\" class=\\"TimelineStatus-eaSVrj fthBdz\\">Cancelled</span><span class=\\"TimelineSubDescription-UGctF iVIATh\\"></span></div>
  </div>
</div>
<div data-testid=\\"solicitation__timeline\\" class=\\"TimelineContainier-bvCAbJ bTSqTF\\">
  <div class=\\"TimelineSubContainier-kqcMpN hzuChq\\">
    <div width=\\"12%\\" class=\\"TimelineQuadrant-fwbGLn TimelineIdentificationQuadrant-bswpNF hBbppX lbXdqa\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"users\\" class=\\"svg-inline--fa fa-users fa-w-20 TimelineIcon-cVZDCs ebPbWO\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 640 512\\" backgroundcolor=\\"#f0f3f7\\" color=\\"#5f6772\\">
        <path fill=\\"currentColor\\" d=\\"M96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm448 0c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm32 32h-64c-17.6 0-33.5 7.1-45.1 18.6 40.3 22.1 68.9 62 75.1 109.4h66c17.7 0 32-14.3 32-32v-32c0-35.3-28.7-64-64-64zm-256 0c61.9 0 112-50.1 112-112S381.9 32 320 32 208 82.1 208 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zm-223.7-13.4C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z\\"></path>
      </svg><span class=\\"TimelineSubDescription-UGctF iVIATh\\">03/24/2020</span></div>
    <div class=\\"TimelineQuadrant-fwbGLn jcrlXN\\"><span class=\\"TimelineLabel-bjBMFt hmpenS\\">ACTION</span><span class=\\"TimelineDesription-AVioK gyAJrn\\">Solicitation approved by Henrique Elias</span></div>
  </div>
  <div class=\\"TimelineSubContainier-kqcMpN kFKWsF\\">
    <div width=\\"15%\\" class=\\"TimelineQuadrant-fwbGLn FbnPL\\"><span class=\\"TimelineLabel-bjBMFt hmpenS\\">STATUS</span><span color=\\"#03a046\\" class=\\"TimelineStatus-eaSVrj ccUBFh\\">Approved</span><span class=\\"TimelineSubDescription-UGctF iVIATh\\"></span></div>
  </div>
</div>
<div data-testid=\\"solicitation__timeline\\" class=\\"TimelineContainier-bvCAbJ bTSqTF\\">
  <div class=\\"TimelineSubContainier-kqcMpN hzuChq\\">
    <div width=\\"12%\\" class=\\"TimelineQuadrant-fwbGLn TimelineIdentificationQuadrant-bswpNF hBbppX lbXdqa\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"asterisk\\" class=\\"svg-inline--fa fa-asterisk fa-w-16 TimelineIcon-cVZDCs ijgjbN\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\" backgroundcolor=\\"#ecf8f8\\" color=\\"#03a046\\">
        <path fill=\\"currentColor\\" d=\\"M478.21 334.093L336 256l142.21-78.093c11.795-6.477 15.961-21.384 9.232-33.037l-19.48-33.741c-6.728-11.653-21.72-15.499-33.227-8.523L296 186.718l3.475-162.204C299.763 11.061 288.937 0 275.48 0h-38.96c-13.456 0-24.283 11.061-23.994 24.514L216 186.718 77.265 102.607c-11.506-6.976-26.499-3.13-33.227 8.523l-19.48 33.741c-6.728 11.653-2.562 26.56 9.233 33.037L176 256 33.79 334.093c-11.795 6.477-15.961 21.384-9.232 33.037l19.48 33.741c6.728 11.653 21.721 15.499 33.227 8.523L216 325.282l-3.475 162.204C212.237 500.939 223.064 512 236.52 512h38.961c13.456 0 24.283-11.061 23.995-24.514L296 325.282l138.735 84.111c11.506 6.976 26.499 3.13 33.227-8.523l19.48-33.741c6.728-11.653 2.563-26.559-9.232-33.036z\\"></path>
      </svg><span class=\\"TimelineSubDescription-UGctF iVIATh\\">03/24/2020</span></div>
    <div class=\\"TimelineQuadrant-fwbGLn jcrlXN\\"><span class=\\"TimelineLabel-bjBMFt hmpenS\\">ACTION</span><span class=\\"TimelineDesription-AVioK gyAJrn\\">Solicitation created by Henrique Elias</span></div>
  </div>
  <div class=\\"TimelineSubContainier-kqcMpN kFKWsF\\"></div>
</div>"
`;
